src/classes/RD2_EntryFormController.cls

```diff
@@ -38,45 +38,100 @@ public with sharing class RD2_EntryFormController {
 
     /**
     * @description  Gather the org setting for Recurring Donation Entry Form
-    * @param parentId The parentId of the new RD
+    * @param parentId The parentId to use when creating the new RD record (if New from Acount or Contact)
     * @return Map<String, Object>
     */
     @AuraEnabled(Cacheable=true)
-    public static Map<String, Object> getSetting(Id parentId) {
+    public static Map<String, Object> getRecurringSettings(Id parentId) {
         String parentSObjectType = (parentId == null)
             ? null
             : parentId.getSobjectType().getDescribe().getName();
 
+        DescribeFieldResult dfrPeriod = UTIL_Describe.getFieldDescribe(
+            'npe03__Recurring_Donation__c', 'npe03__Installment_Period__c'
+        );
+        DescribeFieldResult dfrFrequency = UTIL_Describe.getFieldDescribe(
+            'npe03__Recurring_Donation__c', UTIL_Namespace.StrTokenNSPrefix('InstallmentFrequency__c')
+        );
+
         return new Map<String, Object> {
             'isAutoNamingEnabled' => RD2_NamingService.isAutomaticNamingEnabled,
             'isMultiCurrencyEnabled' => UserInfo.isMultiCurrencyOrganization(),
             'parentSObjectType' => parentSObjectType,
-            'dayOfMonthLastDay' => RD2_Constants.DAY_OF_MONTH_LAST_DAY
+            'InstallmentPeriodPermissions' => new Map<String, Boolean> {
+                    'Createable' => dfrPeriod.isCreateable(),
+                    'Updateable' => dfrPeriod.isUpdateable(),
+                    'Visible' => dfrPeriod.isAccessible()
+                },
+            'InstallmentFrequencyPermissions' => new Map<String, Boolean> {
+                    'Createable' => dfrFrequency.isCreateable(),
+                    'Updateable' => dfrFrequency.isUpdateable(),
+                    'Visible' => dfrFrequency.isAccessible()
+                }
```

Comment:
Extract the Createable, Updateable, and Visible into a helper function that returns Map<String, Boolean>()
public static Map<String, Boolean> getFieldSecurity(String ApiName)

Reply:
Great suggestion